drivers/volume/portworx/portworx.go

```diff
@@ -2974,6 +2982,44 @@ func (d *portworx) ToggleCallHome(n node.Node, enabled bool) error {
 	return nil
 }
 
+func (d *portworx) IsOpearatorBasedInstall() (bool, error) {
+	_, err := apiExtentions.GetCRD("storageclusters.core.libopenstorage.org", metav1.GetOptions{})
+
+	return err == nil, err
+
+}
+
+func (d *portworx) UpdateStorageClusterImage(imageName string) error {
+	pxOps, err := pxOperator.ListStorageClusters(schedops.PXNamespace)
+	if err != nil {
+		er := fmt.Errorf("Error getting Storage Clusters list, Err: %v", err.Error())
+		logrus.Error(er.Error())
+		return er
+
+	}
+
+	stc, err := pxOperator.GetStorageCluster(pxOps.Items[0].Name, pxOps.Items[0].Namespace)
+	if err != nil {
+		er := fmt.Errorf("error getting Storage Clusters [%v], Namespace: [%v], Err: %v", pxOps.Items[0].Name, pxOps.Items[0].Namespace, err.Error())
+		logrus.Error(er.Error())
+		return er
+
+	}
+	logrus.Infof("Current Storage Cluster Image: %v", stc.Spec.Image)
+	stc.Spec.Image = imageName
+	updatedStc, err := pxOperator.UpdateStorageCluster(stc)
+	if err != nil {
+		er := fmt.Errorf("error upgrading Storage Cluster [%v], Namespace: [%v], Err: %v", pxOps.Items[0].Name, pxOps.Items[0].Namespace, err.Error())
+		logrus.Error(er.Error())
+		return er
+
+	}
+	logrus.Infof("Storage Cluster Image updated to [%v]", imageName)
+	logrus.Info(updatedStc)
```

Comment:
I don't think we need to print the entire storage cluster here.

Reply:
fixed